SimTKcommon/Simulation/include/SimTKcommon/internal/State.h

```diff
@@ -1089,6 +1118,41 @@ was created. This has no effect on the realization level.
 **/
 inline void setSystemTopologyStageVersion(StageVersion topoVersion);
 
+/** (Advanced) Return a StageVersion for time, meaning an integer that is
+incremented whenever time is changed. Will be 1 or greater if time has been
+initialized; otherwise, time is meaningless and may be NaN. **/
+inline StageVersion getTimeStageVersion() const;
+/** (Advanced) Return a StageVersion for q, meaning an integer that is
+incremented whenever any q is changed (or more precisely whenever q or y is
+returned with writable access). Will be 1 or greater if q has been
+allocated. **/
+inline StageVersion getQStageVersion() const;
+/** (Advanced) Return a StageVersion for u, meaning an integer that is
+incremented whenever any u is changed (or more precisely whenever u or y is
+returned with writable access). Will be 1 or greater if u has been
+allocated. **/
+inline StageVersion getUStageVersion() const;
+/** (Advanced) Return a StageVersion for z, meaning an integer that is
+incremented whenever any z is changed (or more precisely whenever z or y is
+returned with writable access). Will be 1 or greater if z has been
+allocated. **/
+inline StageVersion getZStageVersion() const;
+
+/** (Advanced) Return a reference to the cache entry information for a 
+particular cache entry. No validity checking is performed. **/
+SimTK_FORCE_INLINE const CacheEntryInfo& 
+getCacheEntryInfo(SubsystemIndex, CacheEntryIndex) const;
+
+/** (Advanced) Return a reference to the discrete variable information for a 
+particular discrete variable. **/
+SimTK_FORCE_INLINE const DiscreteVarInfo& 
+getDiscreteVarInfo(SubsystemIndex, DiscreteVariableIndex) const;
+
+/** (Advanced) Return a reference to the per-subsystem information in the
+state. **/
+SimTK_FORCE_INLINE const PerSubsystemInfo& 
+getPerSubsystemInfo(SubsystemIndex) const;
+
```

Comment:
> How do these all show up in doxygen? Ideally they are towards the bottom or in an advanced group.

Good idea -- I made an advanced group and moved them there. There is still a lot of leftover ugliness in the State doxygen. I don't want to address it now because I have to merge these changes into 4.0 where many other State changes are happening. But at least with an advanced group I'm not making it worse. Thanks.